src/main/java/sparkles/util/Storage.java

```diff
@@ -0,0 +1,134 @@
+package sparkles.util;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import sparkles.SparklesException;
+import sparkles.task.Deadline;
+import sparkles.task.Event;
+import sparkles.task.Task;
+import sparkles.task.Todo;
+
+/**
+ * Represent a storage object, taking in the file path of the task.txt.
+ * Task.txt is a file stored on the local disk
+ * with the tasks that user input.
+ * Deals with loading tasks from the file and saving task in the file.
+ */
+public class Storage {
+
+    protected String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Update task.txt.
+     * Task.txt is overwritten with respect to the
+     * tasks in the list provided as parameter.
+     *
+     * @param storage list of task
+     */
+    public void updateFile(List<Task> storage) {
+        String userDir = new File("").getAbsolutePath();
+        String taskFilePath = userDir + File.separator + filePath;
+        FileWriter fileWriter;
+
+        try {
+            fileWriter = new FileWriter(taskFilePath);
+            StringBuilder append = new StringBuilder();
+            for (Task task : storage) {
+                append.append(task.convertToDiskFormat()).append("\n");
+            }
+
+            fileWriter.write(append.toString());
+            fileWriter.close();
+        } catch (IOException ignored) {
+            System.out.println("Error");
+        }
+
+    }
+
+    /**
+     * Reads task.txt.
+     * Meanwhile, update storage in Tasklist
+     * with the current Tasks in the task.txt.
+     * The text in the .txt is decoded to
+     * differentiate what kind of task(
+     * deadline, event, todo) it is.
+     *
+     * @return a list with the corresponding task
+     * @throws SparklesException
```

Comment:
Perhaps you can write the exception it handles? I.E: throws SparklesException when the file is not found?

Reply:
Forgot to include this, thanks!